---
title: "Figure2"
author: "ML"
date: "2025_02_15"
output: html_document
editor_options: 
  chunk_output_type: console
---
```{r}
setwd("/scratch/Shares/rinn/ML/RNAseq_FULL_MODEL/paper_figures/Fig2")
library(tidyverse)
library(pheatmap)
#library(RcisTarget)
library(WGCNA)
```

#load DESeq2 output and raw data for all cell line
```{r}
base_dir <- "/scratch/Shares/rinn/ML/RNAseq_FULL_MODEL"# Define the base path
#all dds
dds_rdata_files <- list.files(path = base_dir, pattern = "_LRT_res_df.RData$", 
                          full.names = TRUE, recursive = TRUE)# Find all RData files matching the pattern in subdirectories

for (file in dds_rdata_files) {
  load(file)
  cat("Loaded:", file, "\n")
}# Load all matching files

#all DEG
deg_base_dir <- "/scratch/Shares/rinn/ML/RNAseq_FULL_MODEL"

deg_rdata_files <- list.files(path = base_dir, pattern = "_rnaseq_list_for_atacseq.RData$", 
                          full.names = TRUE, recursive = TRUE)

for (file in deg_rdata_files) {
  load(file)
  cat("Loaded:", file, "\n")
}
```

#plot rlog transformed counts heatmap of DEGs of CRNDE
```{r}
load("/scratch/Shares/rinn/ML/RNAseq_FULL_MODEL/g2s.RData")
log_counts <- rlog(CRNDE_dds)

CRNDE_rlog_counts <- log_counts[, grepl("CRNDE", colnames(rlog_counts))]

CRNDE_rlog_counts <- as.data.frame(assay(CRNDE_rlog_counts))

CRNDE_rlog_counts <- CRNDE_rlog_counts %>% rownames_to_column("gene_id") %>%merge(g2s)

filtered_CRNDE_rlog_counts <- CRNDE_rlog_counts[CRNDE_rlog_counts$gene_name %in% CRNDE_trend_sig_filtered$gene_name,]

filtered_CRNDE_rlog_counts_long <- filtered_CRNDE_rlog_counts%>%
  pivot_longer(
    cols = starts_with("CRNDE"),  # Select all columns starting with "CRNDE"
    names_to = c("time_point", "replicate"),  # Create new columns for time point and replicate
    names_pattern = "^CRNDE_(\\d+)_(\\d+)$",  # Use regex to capture time point and replicate
    values_to = "log_counts"  # Name the new column for values
  ) %>%
  mutate(
    time_point = as.numeric(time_point),  # Convert time_point to numeric
    replicate = as.numeric(replicate)  # Convert replicate to numeric
  )

CRNDE_rlog_counts_mean <- filtered_CRNDE_rlog_counts_long %>%
  group_by(gene_name, time_point) %>%
  summarise(mean_log_counts = mean(log_counts), .groups = 'drop')

# Ensure time_point is an ordered factor to preserve the sequence
filtered_CRNDE_rlog_counts_long$time_point <- factor(filtered_CRNDE_rlog_counts_long$time_point, 
                                                     levels = c(0, 2, 4, 8, 16, 24, 48, 96), 
                                                     ordered = TRUE)

# Normalize the data to the 0 timepoint
filtered_CRNDE_rlog_counts_long <- filtered_CRNDE_rlog_counts_long %>%
  group_by(gene_name) %>%
  mutate(normalized_log_counts = log_counts - log_counts[time_point == 0]) %>%
  ungroup()

# Calculate the mean of the normalized log counts per time point
CRNDE_rlog_counts_mean_normalized <- filtered_CRNDE_rlog_counts_long %>%
  group_by(gene_name, time_point) %>%
  summarise(mean_normalized_log_counts = mean(normalized_log_counts), .groups = 'drop')

# Reshape data for heatmap
CRNDE_rlog_counts_matrix <- CRNDE_rlog_counts_mean_normalized %>%
  pivot_wider(names_from = time_point, values_from = mean_normalized_log_counts) %>%
  column_to_rownames("gene_name")


pdf("/scratch/Shares/rinn/ML/RNAseq_FULL_MODEL/paper_figures/Fig2/CRNDE_deg_rlog_heatmap.pdf", width = 5, height = 5)
# Plot heatmap
pheatmap(CRNDE_rlog_counts_matrix, cluster_rows = TRUE, cluster_cols = FALSE, 
         scale = "none", main = "Heatmap of Normalized Gene Expression")

dev.off()

```

#plot log2FC heatmap of DEGs of CRNDE
```{r}
# Add a 0 time_point
  CRNDE_trend_sig_filtered_zero <- CRNDE_trend_sig_filtered %>%
    dplyr::select(gene_id, gene_name, baseMean) %>%
    distinct() %>%
    mutate(log2FoldChange = 0,
           timepoint = 0) 

CRNDE_trend_sig_filtered <- CRNDE_trend_sig_filtered %>%
    bind_rows(CRNDE_trend_sig_filtered_zero)

# converting to matrix for heatmap
CRNDE_trend_sig_filtered_matrix <- CRNDE_trend_sig_filtered %>%
  dplyr::select(gene_name, timepoint, log2FoldChange) %>%
  pivot_wider(names_from = timepoint, names_sort = TRUE, values_from = log2FoldChange) %>%
  column_to_rownames("gene_name") %>%
  as.matrix()

#order
CRNDE_trend_sig_filtered_matrix <- CRNDE_trend_sig_filtered_matrix[,c("0", "2", "4", "8", "16", "24", "48", "96")]

col_pal10 <- colorRampPalette(colors = c("#43071E",
                                         "#691D32",
                                         "#923346",
                                         "#BD4B5C",
                                         "#D17486",
                                         "#E19EB0",
                                         "#F0C5D8",
                                         "#F8F0FE",
                                         "#C8D0EF",
                                         "#98B1DA",
                                         "#6A93C6",
                                         "#4272AE",
                                         "#31508C",
                                         "#1E356C",
                                         "#0E1949"))(100)



pdf("/scratch/Shares/rinn/ML/RNAseq_FULL_MODEL/paper_figures/Fig2/CRNDE_deg_lfc_heatmap.pdf", width = 5, height = 30)

# Generate the heatmap
pheatmap(CRNDE_trend_sig_filtered_matrix, cluster_cols = FALSE, cluster_rows = TRUE, show_rownames = TRUE, scale = "none",
                   color = col_pal10,
                   clustering_distance_rows = "euclidean",
                   clustering_method = "ward.D",
                   border_color = NA,
                   fontsize_row = 5,
                   cellheight = 6,
                   cellwidth = 10,
                   breaks = seq(-5,5, length.out = length(col_pal10)),
                   treeheight_row = 25)

# Close the PDF device
dev.off()

```

##plot tpm heatmap of DEGs of CRNDE
```{r}

```

