---
title: "CRNDE_LRT_FULL_MODEL"
author: "ML"
date: "2025_02_01"
output: html_document
editor_options: 
  chunk_output_type: console
---
```{r setup, include=FALSE}
knitr::opts_chunk$set(echo = TRUE)
options(bitmapType = "cairo")
library(IRanges)
library(tidyverse)
library(magrittr)
library(pheatmap)
library(Rcpp)
library(DESeq2)
library(rmarkdown)
library(MatrixGenerics)
library(ggrepel)
```

#CRNDE LRT
```{r}
setwd("/scratch/Shares/rinn/ML/RNAseq_FULL_MODEL/CRNDE")
#Raw counts file
GFP_counts <- read.table("/scratch/Shares/rinn/lincxpress/GFP_12Sept2024/rnaseq/nextflow_results/star_salmon/salmon.merged.gene_counts.tsv", header=TRUE, row.names=1)
CRNDE_counts <- read.table("/scratch/Shares/rinn/lincxpress/CRNDE/rnaseq/nextflow_results/star_salmon/salmon.merged.gene_counts.tsv", header=TRUE, row.names=1)

#make g2s
g2s <- data.frame(
  gene_id = rownames(GFP_counts),
  gene_name = GFP_counts[, 1]
)
#Remove gene_name
CRNDE_counts <- CRNDE_counts[, -1]
GFP_counts <- GFP_counts[, -1]

# Round counts to integer mode required for DESEQ2
GFP_integer <- round(GFP_counts)
CRNDE_integer <- round(CRNDE_counts)

# Ensure both datasets have the same genes (row order)
all(rownames(GFP_integer) == rownames(CRNDE_integer))


# Combine the GFP and CRNDE counts into a single matrix
combined_counts <- cbind(GFP_integer, CRNDE_integer)

#factor
#condition <- factor(c(rep("GFP", length(GFP_colnames)), rep("CRNDE", length(CRNDE_colnames))))
#time_point <- factor(c(sapply(strsplit(GFP_colnames, "_"), `[`, 2), sapply(strsplit(CRNDE_colnames, "_"), `[`, 2)))
#replicate <- as.factor(c(sapply(strsplit(GFP_colnames, "_"), `[`, 3), sapply(strsplit(CRNDE_colnames, "_"), `[`, 3)))
#col_data <- data.frame(sample_id = colnames(combined_counts), condition = condition, time_point = time_point)

col_data <- data.frame(
  sample_id = colnames(combined_counts))

split_values <- strsplit(col_data$sample_id, "_")

View(split_values)
# So here we will go through each row of split_values and run a "generic function(x)" 
# We will then retain the second item which is the time point value in sample_id
condition_values <- sapply(split_values, function(x) x[[1]])
time_values <- sapply(split_values, function(x) x[[2]])

# Similar to above we are using apply to grab the third fragment in split_values (replicate value)
replicate_values <- sapply(split_values, function(x) x[[3]])

# Adding condition and time point into samplesheet for DESEQ2
col_data$time <- time_values
col_data$condition <- condition_values
# Now let's add another column for replicate
col_data$replicate <- replicate_values

#facotring timepoint column
col_data$replicate <- factor(col_data$replicate, levels = c("1", "2", "3"))
col_data$condition <- factor(col_data$condition, levels = c("GFP", "CRNDE"))
col_data$time <- factor(col_data$time, levels = c("0", "2", "4", "8", "16", "24", "48", "96"))
levels(col_data$time)
levels(col_data$condition)
View(col_data)
```

```{r}
stopifnot(all(colnames(combined_counts) == col_data$sample_id))

# Create the DESeqDataSet
CRNDE_dds <- DESeqDataSetFromMatrix(countData = combined_counts, colData = col_data, design = ~ time+condition+time:condition)
# Perform the LRT to find genes changing due to CRNDE
CRNDE_dds <- DESeq(CRNDE_dds, test = "LRT", reduced = ~ time+condition)

CRNDE_dds <- CRNDE_dds [  rowSums ( counts (CRNDE_dds) ) >  1 , ] 
nrow (CRNDE_dds)
```

```{r}

#Curating all results into data frame
resultsNames(CRNDE_dds)

result_names <- resultsNames(CRNDE_dds)

results_names <- result_names[-1]
View(results_names)

res_df <- data.frame("gene_id" = character(), 
                     "baseMean" = numeric(), 
                     "log2FoldChange" = numeric(), 
                     "lfcSE" = numeric(),
                     "stat" = numeric(),
                     "pvalue" = numeric(),
                     "padj" = numeric(),
                     "gene_name" = character(),
                     "result_name" = character())

# For loop to get all results per time point  

for(i in 1:length(results_names)) {
  results_name <- results_names[i]
  res <- results(CRNDE_dds, name = results_name)
  tmp_res_df <- res %>% as.data.frame() %>%
    rownames_to_column("gene_id") %>%
    merge(g2s) %>%
    mutate(result_name = results_name)
  res_df <- dplyr::bind_rows(res_df, tmp_res_df)
  
}

save(CRNDE_dds, res_df, file = "/scratch/Shares/rinn/ML/RNAseq_FULL_MODEL/CRNDE/results/CRNDE_LRT_res_df.RData")

dynamic_res <- result_names[grepl("conditionCRNDE", result_names)]

# cleaning up each result file
CRNDE_LRT_lfc <- lapply(dynamic_res, function(x) {
  results(CRNDE_dds, 
          name = x) %>%
    as.data.frame() %>%
    rownames_to_column("gene_id") %>% 
    merge(g2s) %>%
    mutate(result_name = x,
           # Corrected the typo here
           timepoint = as.numeric(sub("time([0-9]+)\\.conditionCRNDE", "\\1", result_name)))
}) %>% bind_rows()

  # Calculate the maximum fold-change in any one timepoint
  CRNDE_maxfc <- CRNDE_LRT_lfc %>%
    group_by(gene_id) %>%
    summarize(max_fc = max(abs(log2FoldChange))) 
  
  # merge max shrnklfc into dataframe
  CRNDE_LRT_lfc <- CRNDE_LRT_lfc %>%
    left_join(CRNDE_maxfc)
  
  CRNDE_res_df_padj0.05 <- CRNDE_LRT_lfc %>% 
  filter(padj <= 0.01)
  print(length(unique(CRNDE_res_df_padj0.05$gene_id)))
  #6093
  CRNDE_sig <- CRNDE_res_df_padj0.05 %>%
  filter(baseMean >= 10)
  #6039
  CRNDE_sig <- CRNDE_sig %>%
  filter(max_fc >= 1)
  print(length(unique(CRNDE_sig$gene_id)))
  #1350
  
```

```{r}
CRNDE_sig <- CRNDE_sig %>%
  mutate(
    up = log2FoldChange >= 0 & padj < 0.05,
    down = log2FoldChange <= 0 & padj < 0.05
  )
# Define the relevant time points
valid_time_points <- c(8, 16, 24, 48, 96)

# Filter the data to include only the valid time points
CRNDE_sig_filtered <- CRNDE_sig %>%
  filter(timepoint %in% valid_time_points)

# Create a summary table to count up and down regulations per gene across time points
trend_summary <- CRNDE_sig_filtered %>%
  group_by(gene_id) %>%
  summarize(
    up_count = sum(up, na.rm = TRUE),   # Count TRUE in 'up' for each gene
    down_count = sum(down, na.rm = TRUE), # Count TRUE in 'down' for each gene
    total_comparisons = n()  # Total number of time points per gene
  ) %>%
  filter(total_comparisons == length(valid_time_points))  # Ensure all time points are considered

# Filter for significant up counts from the specific time points (at least 5 time points)
CRNDE_sig_up <- trend_summary %>%
  filter(up_count >= 5)

# Filter for significant down counts from the specific time points (at least 5 time points)
CRNDE_sig_down <- trend_summary %>%
  filter(down_count >= 5)

# Merge back to get the log2FoldChange for plotting
CRNDE_sig <- CRNDE_sig %>%
  filter(gene_id %in% c(CRNDE_sig_up$gene_id, CRNDE_sig_down$gene_id))

# Count the unique number of genes
length(unique(CRNDE_sig$gene_id)) 
#1018
```

```{r}
CRNDE_trend_sig <- CRNDE_sig %>%
  mutate(
    sig_up = log2FoldChange >= 0.58 & padj < 0.01,
    sig_down = log2FoldChange <= -0.58 & padj < 0.01
  )
# Define the relevant time points
valid_time_points_2 <- c(8, 16, 24, 48, 96) 

# Filter the data to include only the valid time points
CRNDE_trend_sig_filtered <- CRNDE_trend_sig %>%
  filter(timepoint %in% valid_time_points_2)

# Create a summary table to count up and down regulations per gene across time points
trend_summary <- CRNDE_trend_sig_filtered %>%
  group_by(gene_id) %>%
  summarize(
    sig_up_count = sum(sig_up, na.rm = TRUE),   # Count TRUE in 'up' for each gene
    sig_down_count = sum(sig_down, na.rm = TRUE), # Count TRUE in 'down' for each gene
    total_comparisons = n()  # Total number of time points per gene
  ) %>%
  filter(total_comparisons == length(valid_time_points_2))  # Ensure all time points are considered

# Filter for significant up counts from the specific time points (at least 5 time points)
CRNDE_trend_sig_up <- trend_summary %>%
  filter(sig_up_count >= 5)

CRNDE_trend_sig_up_filtered <- CRNDE_trend_sig_up %>%
  filter(gene_id %in% CRNDE_trend_sig_up$gene_id)
#151

# Filter for significant down counts from the specific time points (at least 5 time points)
CRNDE_trend_sig_down <- trend_summary %>%
  filter(sig_down_count >= 5)

CRNDE_trend_sig_down_filtered <- CRNDE_trend_sig_down %>%
  filter(gene_id %in% CRNDE_trend_sig_down$gene_id)
#217

CRNDE_trend_sig_down_filtered <- CRNDE_trend_sig_down_filtered %>%
    left_join(CRNDE_sig)
CRNDE_trend_sig_up_filtered <- CRNDE_trend_sig_up_filtered %>%
    left_join(CRNDE_sig)

CRNDE_trend_sig_filtered <- CRNDE_trend_sig %>%
  filter(gene_id %in% c(CRNDE_trend_sig_up_filtered$gene_id, CRNDE_trend_sig_down_filtered$gene_id))

print(length(unique(CRNDE_trend_sig_filtered$gene_id)))
#368
```

#CRNDE sig gene TPM plot
```{r}
#for time model
# Load TPM data for CRNDE
tpm_CRNDE <- read.table("/scratch/Shares/rinn/lincxpress/CRNDE/rnaseq/nextflow_results/star_salmon/salmon.merged.gene_tpm.tsv", header=TRUE, row.names=1)

# Load TPM data for CRNDE
tpm_GFP <- read.table("/scratch/Shares/rinn/lincxpress/GFP_12Sept2024/rnaseq/nextflow_results/star_salmon/salmon.merged.gene_tpm.tsv", header=TRUE, row.names=1)

# Summarize mean TPM for each gene at each time point
mean_tpm_per_time <- tpm_CRNDE %>%
  pivot_longer(cols = starts_with("CRNDE_"), 
               names_to = c("time_point", "replicate"),
               names_pattern = "CRNDE_(\\d+)_(\\d+)",
               values_to = "TPM") %>%
  group_by(gene_name, time_point) %>%
  summarize(mean_TPM = mean(TPM, na.rm = TRUE), .groups = "drop")

# Identify genes where all time points have mean TPM < 0.5
genes_to_exclude <- mean_tpm_per_time %>%
  group_by(gene_name) %>%
  summarize(all_below_threshold = all(mean_TPM < 2), .groups = "drop") %>%
  filter(all_below_threshold) %>%
  pull(gene_name)

# Exclude these genes from your filtered dataset
CRNDE_trend_sig_filtered <- CRNDE_trend_sig_filtered %>%
  filter(!(gene_name %in% genes_to_exclude))
CRNDE_trend_sig_up_filtered <- CRNDE_trend_sig_up_filtered %>%
  filter(gene_id %in% CRNDE_trend_sig_up$gene_id)
CRNDE_trend_sig_down_filtered <- CRNDE_trend_sig_down_filtered %>%
  filter(gene_id %in% CRNDE_trend_sig_down$gene_id)

# Check the number of unique genes remaining
length(unique(CRNDE_trend_sig_filtered$gene_name))
#192

save(CRNDE_LRT_lfc, CRNDE_res_df_padj0.05, CRNDE_sig, CRNDE_trend_sig, CRNDE_trend_sig_down_filtered, CRNDE_trend_sig_up_filtered, CRNDE_trend_sig_filtered, file = "/scratch/Shares/rinn/ML/RNAseq_FULL_MODEL/CRNDE/results/CRNDE_rnaseq_list_for_atacseq.RData")

# Filter the TPM data for genes in sig_CRNDE_LRT_shrnklfc
filtered_tpm_CRNDE <- tpm_CRNDE[tpm_CRNDE$gene_name %in% CRNDE_trend_sig_filtered$gene_name, ]
filtered_tpm_GFP <- tpm_GFP[tpm_GFP$gene_name %in% CRNDE_trend_sig_filtered$gene_name, ]

# Reshape CRNDE TPM data to long format
tpm_long_CRNDE <- filtered_tpm_CRNDE %>%
  pivot_longer(
    cols = starts_with("CRNDE"),
    names_to = c("time_point", "replicate"),
    names_pattern = "^CRNDE_(\\d+)_(\\d+)$",
    values_to = "tpm"
  ) %>%
  mutate(
    time_point = as.numeric(time_point),
    replicate = as.numeric(replicate),
    condition = "CRNDE"  # Add a column to label as CRNDE
  )

# Reshape GFP TPM data to long format
tpm_long_GFP <- filtered_tpm_GFP %>%
  pivot_longer(
    cols = starts_with("GFP"),
    names_to = c("time_point", "replicate"),
    names_pattern = "^GFP_(\\d+)_(\\d+)$",
    values_to = "tpm"
  ) %>%
  mutate(
    time_point = as.numeric(time_point),
    replicate = as.numeric(replicate),
    condition = "GFP"  # Add a column to label as GFP
  )

# Combine CRNDE and GFP data into one data frame
tpm_combined_long <- bind_rows(tpm_long_CRNDE, tpm_long_GFP)

# Calculate mean and standard error (SE) for TPM at each time point and condition
tpm_mean_combined <- tpm_combined_long %>%
  group_by(gene_name, time_point, condition) %>%
  summarise(
    mean_tpm = mean(tpm, na.rm = TRUE),
    se_tpm = sd(tpm, na.rm = TRUE) / sqrt(n()),  # Standard error
    .groups = 'drop'
  )

# Plot the combined TPM values for CRNDE and GFP with error bars
ggplot(tpm_mean_combined, aes(x = time_point, y = mean_tpm, group = interaction(gene_name, condition))) +
  geom_line(aes(color = condition, linetype = condition), alpha = 0.7) +  # Line colored by condition
  geom_point(aes(color = condition), alpha = 0.5) +  # Points for each mean TPM value
  geom_errorbar(
    aes(ymin = mean_tpm - se_tpm, ymax = mean_tpm + se_tpm, color = condition),
    width = 0.2, # Adjust the width of the error bars
    alpha = 0.7
  ) +
  facet_wrap(~ gene_name, scales = "free_y") +  # Separate plots for each gene
  labs(
    x = "Time Point (h)", 
    y = "Mean TPM", 
    color = "Condition"
  ) +
  theme_minimal() +  # Minimal theme
  scale_color_manual(values = c("CRNDE" = "darkred", "GFP" = "blue")) +  # Manual color scale
  scale_linetype_manual(values = c("CRNDE" = "solid", "GFP" = "dashed")) + 
  scale_x_continuous(breaks = unique(tpm_mean_combined$time_point)) +  # Set x-axis breaks
  theme(
    aspect.ratio = 0.6,  # Aspect ratio of 3:2 
    strip.text = element_text(size = 12),  # Adjust facet labels font size if necessary
  )

# Save the plot
ggsave(
  path = "/scratch/Shares/rinn/ML/RNAseq_FULL_MODEL/CRNDE/figures/",
  filename = "CRNDE_GFP_sig_TPM_individule_with_errorbars.pdf",
  height = 20, width = 30, limitsize = FALSE
)
```

#CRNDE sig gene LFC plot
```{r}
load("/scratch/Shares/rinn/ML/RNAseq/GFP_12Sept2024/results/GFP_res_df.RData")
# Filter genes from GFP_res_df that are present in CRNDE_trend_sig_filtered
filtered_GFP_res <- GFP_res_df %>%
  filter(gene_id %in% CRNDE_trend_sig_filtered$gene_id) %>%
  mutate(source = "GFP")

# Add a 0 time_point for GFP data
filtered_GFP_res_zero <- filtered_GFP_res %>%
  dplyr::select(gene_id, gene_name, baseMean) %>%
  distinct() %>%
  mutate(log2FoldChange = 0,
         timepoint = 0,
         source = "GFP")

filtered_GFP_res <- filtered_GFP_res %>%
  bind_rows(filtered_GFP_res_zero)

# Prepare CRNDE data with source column
CRNDE_trend_sig_filtered <- CRNDE_trend_sig_filtered %>%
  mutate(source = "CRNDE")

# Add a 0 time_point for CRNDE
CRNDE_trend_sig_filtered_zero <- CRNDE_trend_sig_filtered %>%
  dplyr::select(gene_id, gene_name, baseMean) %>%
  distinct() %>%
  mutate(log2FoldChange = 0,
         timepoint = 0,
         source = "CRNDE")

CRNDE_trend_sig_filtered <- CRNDE_trend_sig_filtered %>%
  bind_rows(CRNDE_trend_sig_filtered_zero)

# Combine CRNDE and filtered GFP data
combined_data <- bind_rows(CRNDE_trend_sig_filtered, filtered_GFP_res)

# Generate the plot
ggplot(combined_data, aes(x = timepoint, y = log2FoldChange, group = interaction(gene_id, source), color = source)) +
  geom_hline(yintercept = 0, linetype = "dashed") +
  geom_line(alpha = 0.7, aes(linetype = source)) +
  geom_point(alpha = 0.8) +
  facet_wrap(~gene_name, scales = "free_y") +
  scale_color_manual(values = c("CRNDE" = "darkred", "GFP" = "blue")) +
  scale_linetype_manual(values = c("CRNDE" = "solid", "GFP" = "dashed")) + 
  scale_x_continuous(breaks = c(0, 2, 4, 8, 16, 24, 48, 96), labels = c("0", "2", "4", "8", "16", "24", "48", "96")) +
  theme_minimal() +
  labs(title = "Gene LFC Trends for CRNDE and GFP",
       x = "Timepoint (hours)",
       y = "Log2 Fold Change",
       color = "Source",
       linetype = "Source") +
   theme(
    aspect.ratio = 0.6,  # Aspect ratio of 3:2 
    strip.text = element_text(size = 12),  # Adjust facet labels font size if necessary
  )

# Save the plot
ggsave(path = "/scratch/Shares/rinn/ML/RNAseq_FULL_MODEL/CRNDE/figures/", 
       filename = "CRNDE_vs_GFP_trend_sig_lfc_individule.pdf", 
       height = 60, width = 80, limitsize = FALSE)

```

#CRNDE sig gene counts plot
```{r}
# Load counts data for CRNDE
counts_CRNDE <- read.table("/scratch/Shares/rinn/lincxpress/CRNDE/rnaseq/nextflow_results/star_salmon/salmon.merged.gene_counts.tsv", header=TRUE, row.names=1)

# Load counts data for CRNDE
counts_GFP <- read.table("/scratch/Shares/rinn/lincxpress/GFP_12Sept2024/rnaseq/nextflow_results/star_salmon/salmon.merged.gene_counts.tsv", header=TRUE, row.names=1)

# Filter the counts data for genes in sig_CRNDE_LRT_shrnklfc
filtered_counts_CRNDE <- counts_CRNDE[counts_CRNDE$gene_name %in% CRNDE_trend_sig_filtered$gene_name, ]
filtered_counts_GFP <- counts_GFP[counts_GFP$gene_name %in% CRNDE_trend_sig_filtered$gene_name, ]

# Reshape CRNDE counts data to long format
counts_long_CRNDE <- filtered_counts_CRNDE %>%
  pivot_longer(
    cols = starts_with("CRNDE"),
    names_to = c("time_point", "replicate"),
    names_pattern = "^CRNDE_(\\d+)_(\\d+)$",
    values_to = "counts"
  ) %>%
  mutate(
    time_point = as.numeric(time_point),
    replicate = as.numeric(replicate),
    condition = "CRNDE"  # Add a column to label as CRNDE
  )

# Reshape GFP counts data to long format
counts_long_GFP <- filtered_counts_GFP %>%
  pivot_longer(
    cols = starts_with("GFP"),
    names_to = c("time_point", "replicate"),
    names_pattern = "^GFP_(\\d+)_(\\d+)$",
    values_to = "counts"
  ) %>%
  mutate(
    time_point = as.numeric(time_point),
    replicate = as.numeric(replicate),
    condition = "GFP"  # Add a column to label as GFP
  )

# Combine CRNDE and GFP data into one data frame
counts_combined_long <- bind_rows(counts_long_CRNDE, counts_long_GFP)

# Calculate mean and standard error (SE) for counts at each time point and condition
counts_mean_combined <- counts_combined_long %>%
  group_by(gene_name, time_point, condition) %>%
  summarise(
    mean_counts = mean(counts, na.rm = TRUE),
    se_counts = sd(counts, na.rm = TRUE) / sqrt(n()),  # Standard error
    .groups = 'drop'
  )

# Plot the combined counts values for CRNDE and GFP with error bars
ggplot(counts_mean_combined, aes(x = time_point, y = mean_counts, group = interaction(gene_name, condition))) +
  geom_line(aes(color = condition, linetype = condition), alpha = 0.7) +  # Line colored by condition
  geom_point(aes(color = condition), alpha = 0.5) +  # Points for each mean counts value
  geom_errorbar(
    aes(ymin = mean_counts - se_counts, ymax = mean_counts + se_counts, color = condition),
    width = 0.2, # Adjust the width of the error bars
    alpha = 0.7
  ) +
  facet_wrap(~ gene_name, scales = "free_y") +  # Separate plots for each gene
  labs(
    x = "Time Point (h)", 
    y = "Mean counts", 
    color = "Condition"
  ) +
  theme_minimal() +  # Minimal theme
  scale_color_manual(values = c("CRNDE" = "darkred", "GFP" = "blue")) +  
  scale_linetype_manual(values = c("CRNDE" = "solid", "GFP" = "dashed")) + # Manual color scale
  scale_x_continuous(breaks = unique(counts_mean_combined$time_point)) +  # Set x-axis breaks
  theme(
    aspect.ratio = 0.6,  # Aspect ratio of 3:2 
    strip.text = element_text(size = 12),  # Adjust facet labels font size if necessary
  )
  
# Save the plot
ggsave(
  path = "/scratch/Shares/rinn/ML/RNAseq_FULL_MODEL/CRNDE/figures/",
  filename = "CRNDE_GFP_sig_counts_individule_with_errorbars.pdf",
  height = 20, width = 30, limitsize = FALSE
)
```

#volcano plot
```{r}
# Load your DESeq2 results
res <- CRNDE_LRT_lfc 

# Ensure padj values are non-NA
res <- res %>% mutate(padj = replace_na(padj, 1))

# Define your gene list (modify this with your actual gene names)
gene_list <- CRNDE_trend_sig_filtered$gene_name

# Add significance column
res <- res %>%
  mutate(
    significant = case_when(
      padj <= 0.01 & log2FoldChange >= 1  ~ "Upregulated",
      padj <= 0.01 & log2FoldChange <= -1 ~ "Downregulated",
      TRUE ~ "Not Significant"
    )
  )

# Filter genes in gene_list and select only the row with max_fc for each gene
top_genes <- res %>%
  filter(gene_name %in% gene_list) %>%
  group_by(gene_name) %>%
  filter(log2FoldChange == max_fc) %>%  # Keep only the row where log2FC matches max_fc
  ungroup()

# Rank genes by max_fc and keep only the top 20
top_genes <- top_genes %>%
  arrange(desc(max_fc)) %>%  # Sort genes by max_fc in descending order
  slice_head(n = 26)  # Keep only the top 26 genes

# Volcano plot
ggplot(res, aes(x = log2FoldChange, y = -log10(padj), color = significant)) +
  geom_point(alpha = 0.6, size = 2) +
   geom_text_repel(
    data = top_genes, 
    aes(label = gene_name), 
    size = 3, 
    fontface = "bold",
    color = "black",
    box.padding = 0.5,     # Adds space around labels
    point.padding = 0.3,   # Adds space between label and dot
    force = 5,             # Stronger repulsion
    direction = "both",    # Allows movement in all directions
    max.overlaps = Inf
  ) +
  scale_color_manual(values = c("Upregulated" = "red", "Downregulated" = "blue", "Not Significant" = "grey")) +
  labs(title = "Volcano Plot of CRNDE DESeq2 Results",
       x = "Log2 Fold Change",
       y = "-Log10 Adjusted p-value") +
  theme_minimal() +
  theme(legend.title = element_blank()) +
  geom_vline(xintercept = c(-1, 1), linetype = "dashed", color = "black") +
  geom_hline(yintercept = -log10(0.01), linetype = "dashed", color = "black")

ggsave(
  path = "/scratch/Shares/rinn/ML/RNAseq_FULL_MODEL/CRNDE/figures/",
  filename = "CRNDE_sig_volcano.pdf",
  height = 10, width = 10, limitsize = FALSE
)
```
